src/qt/bitcoin.cpp

```diff
@@ -487,13 +487,12 @@ void BitcoinApplication::initializeResult(bool success)
         }
 #endif
 
-        // If -min option passed, start window minimized.
-        if(gArgs.GetBoolArg("-min", false))
-        {
-            window->showMinimized();
-        }
-        else
-        {
+        // If -min option passed, start window minimized or minimized to tray
+        if (gArgs.GetBoolArg("-min", false)) {
+            if (!window->getTrayIconStatus() || !clientModel->getOptionsModel()->getMinimizeToTray()) {
```

Comment:
What if there is a tray icon and `getMinimizeToTray`?

Reply:
@promag 
> What if there is a tray icon and `getMinimizeToTray`?

This is the case described in this PR's description!

With `-min` option and there is a tray icon and `getMinimizeToTray`:
- current master: client hides the splash screen then _shows_ the normal main window (that is an issue);
- this PR: client hides the splash screen then does not show the main window as it is minimized to the tray icon (expected behaviour).